src/auto_scheduler/search_policy/utils.h

```diff
@@ -45,6 +45,27 @@
 namespace tvm {
 namespace auto_scheduler {
 
+/*! \brief Return whether the search task is targeting a CPU. */
+inline bool IsCPUTask(const SearchTask& task) {
+  return ((task)->target->kind->device_type == kDLCPU);
+}
+
+/*! \brief Return whether the search task is targeting a GPU. */
+inline bool IsGPUTask(const SearchTask& task) {
+  return ((task)->target->kind->device_type == kDLGPU ||
+          (task)->target->kind->device_type == kDLOpenCL);
```

Comment:
This function seems to cover more cases
https://github.com/apache/incubator-tvm/blob/e5b793f39fd5b4f84b0aedf06aa376ebe45cf2bc/src/tir/analysis/verify_memory.cc#L151-L154